### PR TITLE
DM-34801: Replace Diff_warpedExp with Diff_templateExp in diaPipe

### PR DESCRIPTION
**Pull request by parejkoj**

### CODE REVIEW

File: pipelines/ApPipe.yaml [hunk 1]
```diff
@@ -65,8 +65,8 @@ contracts:
       transformDiaSrcCat.connections.ConnectionsClass(config=transformDiaSrcCat).diaSourceCat.name
   - imageDifference.connections.ConnectionsClass(config=imageDifference).subtractedExposure.name ==
         diaPipe.connections.ConnectionsClass(config=diaPipe).diffIm.name
-  - imageDifference.connections.ConnectionsClass(config=imageDifference).warpedExposure.name ==
-        diaPipe.connections.ConnectionsClass(config=diaPipe).warpedExposure.name
+  - imageDifference.connections.ConnectionsClass(config=imageDifference).template.name ==
+        diaPipe.connections.ConnectionsClass(config=diaPipe).template.name
```
Comment (kfindeisen) [inline pipelines/ApPipe.yaml:69]: Can the `doWriteWarpedExp` override be removed from `imageDifference` (and likewise in `ApPipeWithFakes`)?

File: python/lsst/ap/pipe/ap_pipe.py [hunk 2]
```diff
@@ -275,7 +275,7 @@ def runAssociation(self, sensorRef):
             solarSystemObjectTable=None,
             diffIm=sensorRef.get(diffType + "Diff_differenceExp"),
             exposure=sensorRef.get("calexp"),
-            warpedExposure=sensorRef.get(diffType + "Diff_warpedExp"),
+            template=sensorRef.get(diffType + "Diff_warpedExp"),
```
Comment (kfindeisen) [inline python/lsst/ap/pipe/ap_pipe.py:278]: Should this be `_templateExp`? Not that I think we've run `ApPipeTask` in a while... 😛

Comment (kfindeisen) [reply python/lsst/ap/pipe/ap_pipe.py:278]: There are also references to `doWriteWarpedExp` in this file.

**Review by kfindeisen — APPROVED** [review]
The changes look reasonable, but please make sure the pipeline is self-consistent about whether or not it's using `warpedExp` anywhere.

File: pipelines/ApPipeWithFakes.yaml [hunk 3]
```diff
@@ -109,8 +109,8 @@ contracts:
       transformDiaSrcCatWithFakes.connections.ConnectionsClass(config=transformDiaSrcCatWithFakes).diaSourceCat.name
   - imageDifferenceWithFakes.connections.ConnectionsClass(config=imageDifferenceWithFakes).subtractedExposure.name ==
       diaPipe.connections.ConnectionsClass(config=diaPipe).diffIm.name
-  - imageDifferenceWithFakes.connections.ConnectionsClass(config=imageDifferenceWithFakes).warpedExposure.name ==
-      diaPipe.connections.ConnectionsClass(config=diaPipe).warpedExposure.name
+  - imageDifferenceWithFakes.connections.ConnectionsClass(config=imageDifferenceWithFakes).template.name ==
+      diaPipe.connections.ConnectionsClass(config=diaPipe).template.name
```
Comment (kherner) [inline pipelines/ApPipeWithFakes.yaml:113]: When I was running with this ticket branch for DM-33911 I was getting an error on BPS submission due to this line. If I instead did

```
imageDifference.connections.ConnectionsClass(config=imageDifference).inputTemplate.name ==
         diaPipe.connections.ConnectionsClass(config=diaPipe).template.name
```
it works.

Comment (kherner) [reply pipelines/ApPipeWithFakes.yaml:113]: Oops, I meant to put that comment in ApPipe.yaml, but this one will presumably have the same problem (change to config=imageDifferenceWithFakes of course).

File: pipelines/ApPipe.yaml [hunk 4]
```diff
@@ -65,8 +65,8 @@ contracts:
       transformDiaSrcCat.connections.ConnectionsClass(config=transformDiaSrcCat).diaSourceCat.name
   - imageDifference.connections.ConnectionsClass(config=imageDifference).subtractedExposure.name ==
         diaPipe.connections.ConnectionsClass(config=diaPipe).diffIm.name
-  - imageDifference.connections.ConnectionsClass(config=imageDifference).warpedExposure.name ==
-        diaPipe.connections.ConnectionsClass(config=diaPipe).warpedExposure.name
+  - imageDifference.connections.ConnectionsClass(config=imageDifference).template.name ==
```
Comment (kherner) [inline pipelines/ApPipe.yaml:68]: When I was running with this ticket branch for DM-33911 I was getting an error on BPS submission due to this line. If I instead did

imageDifference.connections.ConnectionsClass(config=imageDifference).inputTemplate.name ==
         diaPipe.connections.ConnectionsClass(config=diaPipe).template.name
it works.

**Pull request merged**